diff --git a/ksw2_extd2_sse.c b/ksw2_extd2_sse.c
index eafeb4e..1069384 100644
--- a/ksw2_extd2_sse.c
+++ b/ksw2_extd2_sse.c
@@ -2,7 +2,11 @@
 #include <assert.h>
 #include "ksw2.h"
 
-#if defined(__ARM_NEON)
+#if defined(RMINIBWA_USE_SIMDE)
+#define SIMDE_ENABLE_NATIVE_ALIASES
+#include <simde/x86/sse2.h>
+#include <simde/x86/sse4.1.h>
+#elif defined(__ARM_NEON)
 #define __SSE4_1__
 #include "s2n-lite.h"
 #elif defined(__SSE4_1__)
@@ -15,10 +19,10 @@
 
 static inline __m128i ksw_i8x4_to_i32x4(const int8_t *x)
 {
-#if defined(__ARM_NEON)
-    return vreinterpretq_u8_s32(vmovl_s16(vget_low_s16(vmovl_s8(vcreate_s8((uint64_t)(uint32_t)*(int32_t*)x)))));
-#elif defined(__SSE4_1__)
+#if defined(RMINIBWA_USE_SIMDE) || defined(__SSE4_1__)
 	return _mm_cvtepi8_epi32(_mm_cvtsi32_si128(*(int32_t*)x));
+#elif defined(__ARM_NEON)
+    return vreinterpretq_u8_s32(vmovl_s16(vget_low_s16(vmovl_s8(vcreate_s8((uint64_t)(uint32_t)*(int32_t*)x)))));
 #else
 	return _mm_setr_epi32(x[0], x[1], x[2], x[3]);
 #endif
diff --git a/ksw2_extz2_sse.c b/ksw2_extz2_sse.c
index af4981c..89837e7 100644
--- a/ksw2_extz2_sse.c
+++ b/ksw2_extz2_sse.c
@@ -2,7 +2,11 @@
 #include <assert.h>
 #include "ksw2.h"
 
-#if defined(__ARM_NEON)
+#if defined(RMINIBWA_USE_SIMDE)
+#define SIMDE_ENABLE_NATIVE_ALIASES
+#include <simde/x86/sse2.h>
+#include <simde/x86/sse4.1.h>
+#elif defined(__ARM_NEON)
 #define __SSE4_1__
 #include "s2n-lite.h"
 #elif defined(__SSE4_1__)
diff --git a/ksw2_ll_sse.c b/ksw2_ll_sse.c
index 2e4fb31..965b8e9 100644
--- a/ksw2_ll_sse.c
+++ b/ksw2_ll_sse.c
@@ -3,7 +3,10 @@
 #include <string.h>
 #include "ksw2.h"
 
-#if defined(__ARM_NEON)
+#if defined(RMINIBWA_USE_SIMDE)
+#define SIMDE_ENABLE_NATIVE_ALIASES
+#include <simde/x86/sse2.h>
+#elif defined(__ARM_NEON)
 #include "s2n-lite.h"
 #elif defined(__SSE2__)
 #include <xmmintrin.h>
@@ -98,23 +101,23 @@ void *ksw_ll_qinit(void *km, int size, int qlen, const uint8_t *query, int m, co
 
 static inline int ksw_le_u8(__m128i a, __m128i b)
 {
-#if defined(__ARM_NEON)
-	return vmaxvq_u8(vqsubq_u8(a, b)) == 0;
-#elif defined(__SSE2__)
+#if defined(RMINIBWA_USE_SIMDE) || defined(__SSE2__)
 	return _mm_movemask_epi8(_mm_cmpeq_epi8(_mm_subs_epu8(a, b), _mm_setzero_si128())) == 0xffff;
+#elif defined(__ARM_NEON)
+	return vmaxvq_u8(vqsubq_u8(a, b)) == 0;
 #endif
 }
 
 static inline int ksw_max_u8(__m128i x)
 {
-#if defined(__ARM_NEON)
-	return vmaxvq_u8(x);
-#elif defined(__SSE2__)
+#if defined(RMINIBWA_USE_SIMDE) || defined(__SSE2__)
 	x = _mm_max_epu8(x, _mm_srli_si128(x, 8));
 	x = _mm_max_epu8(x, _mm_srli_si128(x, 4));
 	x = _mm_max_epu8(x, _mm_srli_si128(x, 2));
 	x = _mm_max_epu8(x, _mm_srli_si128(x, 1));
 	return _mm_extract_epi16(x, 0) & 0x00ff;
+#elif defined(__ARM_NEON)
+	return vmaxvq_u8(x);
 #endif
 }
 
@@ -223,22 +226,22 @@ end_loop_u8:
 
 static inline int ksw_le_epi16(__m128i a, __m128i b)
 {
-#if defined(__ARM_NEON)
-	return vmaxvq_u16(vcgtq_s16(vreinterpretq_s16_u8(a), vreinterpretq_s16_u8(b))) == 0;
-#elif defined(__SSE2__)
+#if defined(RMINIBWA_USE_SIMDE) || defined(__SSE2__)
 	return _mm_movemask_epi8(_mm_cmpgt_epi16(a, b)) == 0;
+#elif defined(__ARM_NEON)
+	return vmaxvq_u16(vcgtq_s16(vreinterpretq_s16_u8(a), vreinterpretq_s16_u8(b))) == 0;
 #endif
 }
 
 static inline int ksw_max_i16(__m128i x)
 {
-#if defined(__ARM_NEON)
-	return vmaxvq_s16(vreinterpretq_s16_u8(x));
-#elif defined(__SSE2__)
+#if defined(RMINIBWA_USE_SIMDE) || defined(__SSE2__)
 	x = _mm_max_epi16(x, _mm_srli_si128(x, 8));
 	x = _mm_max_epi16(x, _mm_srli_si128(x, 4));
 	x = _mm_max_epi16(x, _mm_srli_si128(x, 2));
 	return _mm_extract_epi16(x, 0);
+#elif defined(__ARM_NEON)
+	return vmaxvq_s16(vreinterpretq_s16_u8(x));
 #endif
 }
 
