--- a/kalloc.c
+++ b/kalloc.c
@@ -219,6 +219,11 @@
 {
 	km_stat_t st;
 	km_stat(km, &st);
+#if defined(__MINGW32__) || defined(__MINGW64__)
+	fprintf(stderr, "[km_stat] cap=%zu, avail=%zu, largest=%zu, n_core=%zu, n_block=%zu\n",
+			st.capacity, st.available, st.largest, st.n_cores, st.n_blocks);
+#else
 	fprintf(stderr, "[km_stat] cap=%ld, avail=%ld, largest=%ld, n_core=%ld, n_block=%ld\n",
 			st.capacity, st.available, st.largest, st.n_blocks, st.n_cores);
+#endif
 }
