--- a/kthread.c
+++ b/kthread.c
@@ -49,6 +49,9 @@ static void *ktf_worker(void *data)
 	while ((i = steal_work(w->t)) >= 0)
 		w->t->func(w->t->data, i, w - w->t->w);
 	pthread_exit(0);
+#if defined(__MINGW32__) || defined(__MINGW64__)
+	return 0;
+#endif
 }
 
 void kt_for(int n_threads, void (*func)(void*,long,int), void *data, long n)
@@ -125,6 +128,9 @@ static void *ktp_worker(void *data)
 		pthread_mutex_unlock(&p->mutex);
 	}
 	pthread_exit(0);
+#if defined(__MINGW32__) || defined(__MINGW64__)
+	return 0;
+#endif
 }
 
 void kt_pipeline(int n_threads, void *(*func)(void*, int, void*), void *shared_data, int n_steps)
